sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereLogic.cs

```diff
@@ -54,77 +61,77 @@ public WhereLogic Contains(string value, string[] searched)
         /// </summary>
         /// <param name="expression"> The expression being checked for null. </param>
-        /// <returns> ADT query that already contains SELECT and FROM. </returns>
-        public WhereLogic IsNull(string expression)
+        /// <returns> Logical operator to combine different WHERE functions or conditions. </returns>
+        public LogicalOperator IsNull(string expression)
         {
-            _clauses.Add(new WhereClause($"{QueryConstants.IsNull}({expression})"));
-            return this;
+            _conditions.Append(new WhereClause($"{QueryConstants.IsNull}({expression})").Condition);
+            return _logicalOperator;
         }
 
         /// <summary>
         /// Adds the <see href="https://docs.microsoft.com/en-us/azure/digital-twins/reference-query-functions#startswith">STARTSWITH</see> function to the condition statement of the query.
         /// </summary>
         /// <param name="stringToCheck"> String to check the beginning of. </param>
         /// <param name="beginningString"> String representing the beginning to check for. </param>
-        /// <returns> ADT query that already contains SELECT and FROM. </returns>
-        public WhereLogic StartsWith(string stringToCheck, string beginningString)
+        /// <returns> Logical operator to combine different WHERE functions or conditions. </returns>
+        public LogicalOperator StartsWith(string stringToCheck, string beginningString)
         {
-            _clauses.Add(new WhereClause($"{QueryConstants.StartsWith}({stringToCheck}, '{beginningString}')"));
-            return this;
+            _conditions.Append(new WhereClause($"{QueryConstants.StartsWith}({stringToCheck}, '{beginningString}')").Condition);
+            return _logicalOperator;
         }
 
         /// <summary>
         /// Adds the <see href="https://docs.microsoft.com/en-us/azure/digital-twins/reference-query-functions#endswith">ENDSWITH</see> function to the condition statement of the query.
         /// </summary>
         /// <param name="stringToCheck"> String to check the ending of. </param>
         /// <param name="endingString"> String representing the ending to check for. </param>
-        /// <returns> ADT query that already contains SELECT and FROM. </returns>
-        public WhereLogic EndsWith(string stringToCheck, string endingString)
+        /// <returns> Logical operator to combine different WHERE functions or conditions. </returns>
+        public LogicalOperator EndsWith(string stringToCheck, string endingString)
         {
-            _clauses.Add(new WhereClause($"{QueryConstants.EndsWith}({stringToCheck}, '{endingString}')"));
-            return this;
+            _conditions.Append(new WhereClause($"{QueryConstants.EndsWith}({stringToCheck}, '{endingString}')").Condition);
```

Comment:
nit comment: You are printing the parenthesis here but on line 173 you use the constant to print the open and close parenthesis ... I think you can get away with not having parenthesis as constants, but if you do, you have to use them everywhere you open or close them.